File: src/main/java/com/cobblegyms/util/MessageUtil.java
package com.cobblegyms.util;

import net.minecraft.network.packet.s2c.play.SubtitleS2CPacket;
import net.minecraft.network.packet.s2c.play.TitleS2CPacket;
import net.minecraft.server.MinecraftServer;
import net.minecraft.server.network.ServerPlayerEntity;
import net.minecraft.text.Text;
import net.minecraft.util.Formatting;

import com.cobblegyms.model.PokemonType;

public class MessageUtil {

    private static final String PREFIX = "\u00a77[\u00a76CobbleGyms\u00a77] \u00a7r";

    public static void sendSuccess(ServerPlayerEntity player, String message) {
        player.sendMessage(Text.literal(PREFIX + "\u00a7a" + message), false);
    }

    public static void sendError(ServerPlayerEntity player, String message) {
        player.sendMessage(Text.literal(PREFIX + "\u00a7c" + message), false);
    }

    public static void sendInfo(ServerPlayerEntity player, String message) {
        player.sendMessage(Text.literal(PREFIX + "\u00a77" + message), false);
    }

    public static void sendWarning(ServerPlayerEntity player, String message) {
        player.sendMessage(Text.literal(PREFIX + "\u00a7e" + message), false);
    }

    public static void sendTitle(ServerPlayerEntity player, String title, String subtitle) {
        player.networkHandler.sendPacket(new TitleS2CPacket(Text.literal(title)));
        if (subtitle != null && !subtitle.isEmpty()) {
            player.networkHandler.sendPacket(new SubtitleS2CPacket(Text.literal(subtitle)));
        }
    }

    public static void broadcast(MinecraftServer server, String message) {
        server.getPlayerManager().broadcast(Text.literal(PREFIX + message), false);
    }

    public static void broadcastColored(MinecraftServer server, String message) {
        server.getPlayerManager().broadcast(Text.literal(PREFIX + message), false);
    }

    public static String formatType(PokemonType type) {
        if (type == null) return "\u00a77Unknown";
        return type.getColorCode() + type.getDisplayName();
    }

    public static String colorize(String text) {
        return text.replace("&", "\u00a7");
    }

    public static String stripColor(String text) {
        return text.replaceAll("\u00a7[0-9a-fk-or]", "");
    }
}

File: src/main/kotlin/com/cobblegyms/util/MessageUtil.kt
package com.cobblegyms.util

import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.text.ClickEvent
import net.minecraft.text.HoverEvent
import net.minecraft.text.Style
import net.minecraft.text.Text

/**
 * Utility class for sending formatted messages to players.
 */
object MessageUtil {
    
    fun sendSuccess(player: ServerPlayerEntity, message: String) {
        player.sendMessage(Text.literal("§a✓ $message"))
    }
    
    fun sendError(player: ServerPlayerEntity, message: String) {
        player.sendMessage(Text.literal("§c✗ $message"))
    }
    
    fun sendWarning(player: ServerPlayerEntity, message: String) {
        player.sendMessage(Text.literal("§e⚠ $message"))
    }
    
    fun sendInfo(player: ServerPlayerEntity, message: String) {
        player.sendMessage(Text.literal("§7ℹ $message"))
    }
    
    fun sendTitle(player: ServerPlayerEntity, title: String) {
        player.sendMessage(Text.literal("§6§l$title"))
    }
    
    fun sendDivider(player: ServerPlayerEntity) {
        player.sendMessage(Text.literal("§8§m──────────────────────────────"))
    }
    
    fun sendClickable(player: ServerPlayerEntity, text: String, command: String, hover: String) {
        val style = Style.EMPTY
            .withClickEvent(ClickEvent(ClickEvent.Action.RUN_COMMAND, command))
            .withHoverEvent(HoverEvent(HoverEvent.Action.SHOW_TEXT, Text.literal(hover)))
        player.sendMessage(Text.literal(text).setStyle(style))
    }
    
    fun formatTime(seconds: Long): String = TimeUtil.formatDuration(seconds)
    
    fun formatWinrate(wins: Int, battles: Int): String {
        if (battles == 0) return "0%"
        return "${String.format("%.1f", wins.toDouble() / battles * 100)}%"
    }
    
    /**
     * Broadcast a message to all online players.
     */
    fun broadcast(server: net.minecraft.server.MinecraftServer, message: String) {
        server.playerManager.playerList.forEach { player ->
            player.sendMessage(Text.literal(message))
        }
    }
    
    /**
     * Broadcast a message to all players with a specific permission level.
     */
    fun broadcastToOps(server: net.minecraft.server.MinecraftServer, message: String) {
        server.playerManager.playerList
            .filter { server.playerManager.isOperator(it.gameProfile) }
            .forEach { it.sendMessage(Text.literal(message)) }
    }
}
